### PR TITLE
Added random function

### PR DESCRIPTION
**Pull request by mrkbac**
This PR add a `random()` function.

Example output:

```nbt
>>> 10 * random() // floor

  floor(10 × random())

    = 6 
```

Let me know if I should change anything!

### CODE REVIEW

File: numbat/tests/interpreter.rs [hunk 1]
```diff
@@ -362,6 +362,7 @@ fn test_other_functions() {
     expect_output("floor(3.9)", "3");
     expect_output("round(3.9)", "4");
     expect_output("round(3.1)", "3");
+    expect_output("random() == random()", "false");
```
Comment (mhmdanas) [inline numbat/tests/interpreter.rs:365]: This test could fail in theory (though very unlikely), so not sure we should have it.

Comment (mhmdanas) [reply numbat/tests/interpreter.rs:365]: Actually this is probably so unlikely that we can probably disregard it entirely, /shrug

File: numbat/src/ffi.rs [hunk 2]
```diff
@@ -903,3 +912,11 @@ fn from_unixtime(args: &[Value]) -> Result<Value> {
 
     Ok(Value::DateTime(dt))
 }
+
+fn random(args: &[Value]) -> Result<Value> {
+    assert!(args.len() == 0);
+
+    let output = fastrand::f64();
```
Comment (mhmdanas) [inline numbat/src/ffi.rs:919]: We could use the `rand` crate instead (it's already a transitive dependency of Numbat).

**Review by mhmdanas — CHANGES_REQUESTED** [review]

File: numbat/src/ffi.rs (hunk 2 above)
Comment (mrkbac) [reply numbat/src/ffi.rs]: I've changed it to `rand` and removed the test.

**Review by sharkdp — APPROVED** [review]
Thank you!

**Comment by Bzero** [comment]
I am just a minute too late with my comment but I think it might be worth to think about if the `rand()` function should take `min` and `max` arguments. 

For many applications one will need to scale the default [0, 1) interval and to me `rand(a, b)` looks more convenient than `rand() * (b - a) + a`.

**Comment by mhmdanas** [comment]
We don't have optional parameters at the moment so we might want to leave it until that's implemented.

**Comment by sharkdp** [comment]
@Bzero Thank you for your comment. I had similar thoughts before merging. But I was hoping that we could now implement a function like rand(a, b) or randint(from, to) using that one source of randomness `random()`. If that turns out to be tricky (e.g. because of subtle biases) , let's add a more FFI functions or rethink this one.

**Comment by Bzero** [comment]
Yes that sounds reasonable, it would probably make sense to implement these features in a module in numbat itself.

**Comment by sharkdp** [comment]
> to implement these features in a module in numbat itself

Yes, absolutely. That's what I meant. Should have mentioned it. In general, I try to keep the FFI as small as possible.

**Pull request merged**